fix(2021.1): remove hard coded links to the production documentation site

vbotnari12 opened the pull request:
- remove hard coded links to documentation.bonitasoft.com and use xref instead
- adding bcd branch when building the preview to make xref validation pass

vbotnari12 committed:
Update build-pr-preview.yml

tbouffard commented on .github/workflows/build-pr-preview.yml:
```diff
@@ -14,9 +14,19 @@ jobs:
+            --component "${{ env.COMPONENT_NAME }}" 
+            --branch "${{ env.COMPONENT_BRANCH_NAME }}"
+            --start-page "${{ env.COMPONENT_VERSION }}"@"${{ env.COMPONENT_NAME }}"::index.adoc
+            --component-with-branches bcd:"${{ env.BCD_BRANCH }}"
```
ℹ️ the bonita component and the branch of this PR are not built anymore 😸

abirembaut replied:
@tbouffard so, this means updating this file in this PR is useless ?

tbouffard replied:
Updating this file is mandatory to make the build pass.

We are introducing xref to bcd 3.6, so the bcd 3.6 branch must be part of the build preview in addition to the branch of this PR of the bonita component.
Have a look at https://github.com/bonitasoft/bonita-doc/blob/2021.2/.github/workflows/build-pr-preview.yml for instance to know how it should look like. 
Feel free to contact me with DM for more information.

abirembaut replied:
Thanks for the info
done
Let me know if I misunderstood

tbouffard replied:
It looks good. Thanks

For now, having to configure all these components and branches is a pain. We are working on a solution to totally drop it and so, simplify the maintenance. See https://github.com/bonitasoft/bonita-documentation-site/issues/326 for more details

tbouffard reviewed (commented):
ℹ️ when introducing links to bcd and other versioned component, please introduce AsciiDoc attributes like what is done in the bcd doc https://github.com/bonitasoft/bonita-continuous-delivery-doc/blob/3.6/antora.yml. The attributes can be used in the version part of the xref and make maintenance easier. See 44277c8
For instance, if in 2023.1 we want to ref test-tookit 1.1, this will be done only by updating the attribute instead of updating all xref.

```yaml
asciidoc:
  attributes:
    bonitaDocVersion: '2022.1'
    testToolkitVersion: '1.0'
```

tbouffard committed:
introduce bcd doc version asciidoc attribute + use xref in automating-builds.adoc

tbouffard committed:
use direct xref in languages.adoc

tbouffard committed:
use direct xref in ldap.adoc instead of wrong javadoc link

Was correct in ad6e19bede732baed66ff65463897ab7ad583d17 but was overridden by a merge commit in
0d7afd0102f6ade6806a7fc35a567c9e726b88ca

tbouffard committed:
use direct xref in software-extensibility.adoc

abirembaut reviewed (approved):
LGTM for the links updates.
I just have a question for Thomas to know if the changes in build-pr-preview.yml need to be reverted

tbouffard reviewed (changes requested):
They are some remaining pages that still use hard codes url and the PR build is broken.

abirembaut committed:
Update build-pr-preview.yml

tbouffard committed:
use direct xref in comparison-of-7-x-and-6-x.adoc

tbouffard committed:
use direct xref in set-up-continuous-integration.adoc

tbouffard committed:
use direct xref in user-application-overview.adoc

tbouffard committed:
use direct xref in admin-application-overview.adoc

tbouffard committed:
use direct xref in bonita-studio-download-installation.adoc

tbouffard committed:
use direct xref in create-application.adoc

tbouffard committed:
use direct xref in creating-an-actor-filter.adoc

tbouffard committed:
how-a-bdm-is-deployed.adoc: add a comment to justify usage of doc site url

tbouffard commented on modules/ROOT/pages/comparison-of-7-x-and-6-x.adoc:
```diff
@@ -127,14 +127,15 @@ See the xref:bonita-bpm-installation-overview.adoc[Installation overview].
 
 | Anonymous user
+// TODO find the xref for "anonymous user"
 | You can now complete a task as an http://documentation.bonitasoft.com/anonymous-user[anonymous user], that is, without being registered in the organization. For example, on an e-commerce site, a new user can browse stock and save items to a basket, then register with the site if they want to save their basket for later or to buy something
```
@abirembaut do you know if we have a page to redirect to?

abirembaut replied:
it no longer exists but I added the link to the guest user feature in the 7.x column that is the equivalent in 7.x

tbouffard commented on modules/ROOT/pages/comparison-of-7-x-and-6-x.adoc:
```diff
@@ -74,7 +74,7 @@ See the xref:bonita-bpm-installation-overview.adoc[Installation overview].
 | Improved UI
 | Not in 6.x
-| Improvements to the Details panel so that tab structure reflects typical worksflow
+| Improvements to the Details panel so that tab structure reflects typical workflow
```
```suggestion
| Improvements to the Details panel so that tab structure reflects typical workflows
```

tbouffard reviewed (commented):
I have fixed most of the hard coded links. In some pages I have updated, I also fix minor formatting issues and typos.

tbouffard committed:
comparison-of-7-x-and-6-x.adoc: fix typo

tbouffard commented on modules/ROOT/pages/user-application-overview.adoc:
```diff
@@ -31,7 +31,7 @@ After Studio validation, the application is imported or cloned; you can view its
 == Access in pre-Production and in Production
 
-To deploy the applications into a bundle or the Cloud, you can use https://documentation.bonitasoft.com/bcd//_manage_living_application[Bonita Continuous Delivery] (for Enterprise, Performance, Efficiency, and Teamwork editions, and using BCD).
+To deploy the applications into a bundle or the Cloud, you can use {bcdDocVersion}@bcd::livingapp_manage_configuration.adoc[Bonita Continuous Delivery] (for Enterprise, Performance, Efficiency, and Teamwork editions, and using BCD).
```
❓ link to bcd_cli?

abirembaut replied:
done

tbouffard commented on modules/ROOT/pages/admin-application-overview.adoc:
```diff
@@ -37,7 +37,7 @@ After Studio validation, the application is imported or cloned; you can view its
 == Access in pre-Production and in Production
 
-To deploy the applications into a bundle or the Cloud, you can use https://documentation.bonitasoft.com/bcd//_manage_living_application[Bonita Continuous Delivery] (for Enterprise, Performance, Efficiency, and Teamwork editions, and using BCD).
+To deploy the applications into a bundle or the Cloud, you can use {bcdDocVersion}@bcd::livingapp_manage_configuration.adoc[Bonita Continuous Delivery] (for Enterprise, Performance, Efficiency, and Teamwork editions, and using BCD).
```
❓ link to bcd_cli?

abirembaut replied:
done

abirembaut committed:
Update comparison-of-7-x-and-6-x.adoc

abirembaut committed:
Update software-extensibility.adoc

abirembaut committed:
Update user-application-overview.adoc

abirembaut committed:
Update user-application-overview.adoc

tbouffard committed:
admin-application-overview.adoc: xref to bcd_cli

tbouffard reviewed (approved):
LGTM thanks @abirembaut
